docs/reference/docs/data-replication.asciidoc

```diff
@@ -0,0 +1,136 @@
+. For each relevant shard, select an active copy from the shard replication group. This can be either the primary or
+   a replica. By default, Elasticsearch will simply round robin between the shard copies.
+. Send shard level read requests to the selected copies.
+. Combine the results and respond. Note that in the case of get by ID look up, only one shard is relevant and this step can be skipped.
+
+==== Failure handling
+
+When a shard fails to respond to a read request, the coordinating node will select another copy from the same replication group
+and send the shard level search request to that copy instead. Repetitive failures can result in no shard copies being available.
+In such cases Elasticsearch will prefer to respond fast, albeit with partial results, instead of waiting for the issue to be
+resolved (partial results are indicated in the `_shards` header of the response).
+
+=== A few simple implications
+
+Each of these basic flows determines how Elasticsearch behaves as a system for both reads and writes. Furthermore, since read
+and write requests can be executed concurrently, these two basic flows interact with each other. This has a few inherent implications:
+
+Efficient reads:: under normal operation each read operation is performed once for each relevant replication group.
+   Only under failure conditions do multiple copies of the same shard execute the same search.
+
+Read uncommitted:: since the primary first indexes locally and then replicates the request, it is possible for a
+  concurrent read to already see the change before it has been acknowledged.
+
+Two copies by default:: this model can be fault tolerant while maintaining only two copies of the data. This is in contrast to
+  quorum based system where the minimum number of copies for fault tolerance is 3.
+
+=== Failures
+
+Under failures, the following is possible:
+
+A single shard can slow down indexing:: since the primary waits on all shards in the in-sync replica set during each operation,
+  a single slow shard can slow down the entire replication group. This is the price we pay for the read efficiency mentioned above.
+  Of course a single slow shard will also slow down unlucky searches that have been routed to it.
+
+Dirty reads:: An isolated primary can expose writes that will not be committed. This is caused by the fact that an isolated
+  primary will only realize that it is isolated once it sends requests to its replicas or when reaching out to the master.
+  At that point the operation is already indexed into the primary and can be read by a concurrent read. Elasticsearch mitigates
+  this risk by pinging the master every second (by default) and rejecting indexing operations if no master is known.
+
+
+=== The Tip of the Iceberg
+
+This document gives a high level overview of how things work. It is meant to give people a  general understanding of how Elasticsearch
```

Comment:
This document gives a high level overview of how things work. It is meant to give people a  general understanding of how Elasticsearch deals with data.

->

This document provides a high level overview of how Elasticsearch deals with data.